### ISSUE TITLE
Perk drops

### ISSUE THREAD

**Issue by Phil25** [Phil25/RTD#75]
Whole new feature which allows good perks to be dropped on the ground and shared with other players.

## TL;DR

* Only good and timed perks are droppable.
* Perks are dropped on death (ConVar: `sm_rtd2_perk_drop_death`).
* Perks can be dropped with `dropitem` command (intel prioritized first, ConVar: `sm_rtd2_perk_drop_self`).
* Dropped perks disappear after 10 seconds.
* Dropped perks save their remaining time and restart once picked up.
* Dropped perks show annotation to nearby players telling them the perk's name and remaining time.
* Picking up a perk requires a player to stand in its pickup range for a few seconds.

## Overview

Encourage teamwork even more by giving people a chance to drop their current perk via `dropitem` command (default `L`). This will be a very obscure feature by itself and people won't know about it since there really isn't a good way of informing this without a message, which RTD is not going to print.

## Visuals & Audio

* Dropped perks are indicated by some particle effect.
* Upon getting close, a notification appears showing the perks name and remaining time (ex. `PowerPlay (8s)`).
* Upon standing in pickup range, picking up starts indicated by a PASS time beam linking the player and the effect. Some loading/charging-sounding audio FX starts playing.
* After a few seconds of standing in pickup range, some equip sound plays and perk is applied for the remaining time.[
